tests/test_transform.py

```diff
@@ -0,0 +1,151 @@
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+
+import numpy as np
+
+import modmesh as mm
+
+
+class TransformTB(mm.testing.TestBase):
+
+    def setUp(self):
+        pass
+
+    def mm_complex(self, real=0.0, imag=0.0):
+        pass
+
+    def real_rng(self):
+        pass
+
+    def imag_rng(self):
+        pass
+
+    def mm_simplearraycomplex(self, size):
+        pass
+
+    def test_numpy_dft_comparison(self):
+        input_size = 100
+
+        mm_input = self.mm_simplearraycomplex(input_size)
+        for i in range(input_size):
+            mm_input[i] = self.mm_complex(self.real_rng(), self.imag_rng())
+
+        np_input = np.array(mm_input, copy=False)
+
+        mm_output = self.mm_simplearraycomplex(input_size)
+        mm.FourierTransform.dft(mm_input, mm_output)
+
+        np_output = np.fft.fft(np_input)
+
+        for i in range(input_size):
+            self.assert_allclose(mm_output[i].real, np_output[i].real)
+            self.assert_allclose(mm_output[i].imag, np_output[i].imag)
+
+    def test_numpy_fft_comparison(self):
+        input_size = 100
+
+        mm_input = self.mm_simplearraycomplex(input_size)
+        for i in range(input_size):
+            mm_input[i] = self.mm_complex(self.real_rng(), self.imag_rng())
+
+        np_input = np.array(mm_input, copy=False)
+
+        mm_output = self.mm_simplearraycomplex(input_size)
+        mm.FourierTransform.fft(mm_input, mm_output)
+
+        np_output = np.fft.fft(np_input)
+
+        for i in range(input_size):
+            self.assert_allclose(mm_output[i].real, np_output[i].real)
+            self.assert_allclose(mm_output[i].imag, np_output[i].imag)
+
+    def test_numpy_ifft_comparison(self):
+        input_size = 100
+
+        mm_input = self.mm_simplearraycomplex(input_size)
+        for i in range(input_size):
+            mm_input[i] = self.mm_complex(self.real_rng(), self.imag_rng())
+
+        np_input = np.array(mm_input, copy=False)
+
+        mm_output = self.mm_simplearraycomplex(input_size)
+        mm.FourierTransform.ifft(mm_input, mm_output)
+
+        np_output = np.fft.ifft(np_input)
+
+        for i in range(input_size):
+            self.assert_allclose(mm_output[i].real, np_output[i].real)
+            self.assert_allclose(mm_output[i].imag, np_output[i].imag)
+
+
+class TransformFp32TC(TransformTB, unittest.TestCase):
+
+    def assert_allclose(self, *args, **kw):
+        if 'atol' not in kw:
+            kw['atol'] = 1.e-2
+        return super().assert_allclose(*args, **kw)
+
+    def mm_complex(self, real=0.0, imag=0.0):
```

Comment:
The multitude of test helpers are bad for maintainability.  Please move the helper functions into types in `setUp`:

```python
def setUp(self):
    self.complex = mm.complex64
    self.SimpleArray = mm.SimpleArrayComplex64
    # so on and so forth.
```